lcls_naming_tool/lcls_naming_tool.py

```diff
@@ -0,0 +1,309 @@
+
+        third_element = name[2]
+
+        # check for name with 3 elements (FFFFF:GGG:CCC, FFFFF:CCC:NN, FFFFF:CCC:XXXX)
+        if (len(name) == 3):
+            fg_valid = fungible_is_valid(second_element)
+            ccc_valid = constituent_component_is_valid(second_element)
+
+            if fg_valid:   
+                ccc_valid = constituent_component_is_valid(third_element)
+                if ccc_valid:
+                    print('Valid')
+                    return True
+                else:
+                    print('Invalid')
+                    return False
+            elif ccc_valid:
+                    element_valid = starts_alphanumeric(third_element)
+                    if element_valid:
+                        print('Valid')
+                        return True
+                    else:
+                        print('Invalid')
+                        return False
+            else:
+                print('Invalid')
+                return False
+
+        fourth_element = name[3]
+
+        # Check for name with 4 elements (FFFFF:GGG:CCC:NN, FFFFF:GGG:CCC:XXXX, FFFFF:CCC:NN:XXXX)
+        if (len(name) == 4):
+            fg_valid = fungible_is_valid(second_element)
+            ccc_valid = constituent_component_is_valid(second_element)
+
+            if fg_valid:
+                ccc_valid = constituent_component_is_valid(third_element)
+                fourth_element_valid = starts_alphanumeric(fourth_element)
+
+                if ccc_valid and fourth_element_valid:
+                    print('Valid')
+                    return True
+                else:
+                    print('Invalid')
+                    return False
+                
+            elif ccc_valid:
+                increment_valid = increment_is_valid(third_element)
+                cspv_valid = starts_alphanumeric(fourth_element)
+
+                if increment_valid and cspv_valid:
+                    print('Valid')
+                    return True
+                else:
+                    print('Invalid')
+                    return False
+                
+            else:
+                print('Invalid')
+                return False
+
+        # check for name with 5 elements (FFFFF:GGG:CCC:NN:XXXX)
+        if (len(name) == 5):
+            fg_taxon = name[1]
+            fg_valid = fungible_is_valid(fg_taxon)
+
+            ccc_taxon = name[2]
+            ccc_valid = constituent_component_is_valid(ccc_taxon)
+
+            nn_taxon = name[3]
+            nn_valid = increment_is_valid(nn_taxon)
+
+            fifth_element = name[4]
+            fifth_element_valid = starts_alphanumeric(fifth_element)
+
+            if fg_valid and ccc_valid and nn_valid and fifth_element_valid:
+                print('Valid')
+                return True
+            else:
+                print('Invalid')
+                return False
+
+    else:
+        print('Invalid')
+        return False
+
+
+def main():
+
+    display_version()
```

Comment:
Usually version reporting in cli is a separate path that doesn't also do the main work of the program. It would be pretty normal to make `main` exit right here if someone asks to display the version.

Reply:
Updated display_version() to call exit(). I think this section will also be re-written with argparse.